discovery-infra/test_infra/controllers/proxy_controller/proxy_controller.py

```diff
@@ -0,0 +1,71 @@
+import os
+import shutil
+import socket
+import logging
+from jinja2 import Environment, PackageLoader
+
+from test_infra import utils
+
+PROXY_USER = 'assisted'
+PROXY_USER_PASS = 'redhat'
+
+class ProxyController():
+    def __init__(self, name=None, port=None, denied_port=None, authenticated=False):
+        if not name:
+            self.address = ''
+        else:
+            self.name = name
+            self.port = port
+            self.authenticated = authenticated
+            self.image = 'docker.io/sameersbn/squid'
+            self._pull_image()
+            self._set_server_address()
+            self._create_conf_from_template(denied_port=denied_port)
+            self._create_user_file_for_auth()
+            self._run_proxy_server()
+
+    def remove(self):
+        if self.address:
+            logging.info(f'Removing Proxy Server {self.name}')
+            utils.remove_running_container(self.name)
+            self._remove_config()
+
+    def _pull_image(self):
+        logging.info('Pulling Proxy Server Image')
+        utils.pull_container_image(self.image)
```

Comment:
why u need to pull image? not enough just to run and give podman to do it?

Reply:
wanted to be explicit about it. It is enough to let podman do it
will change

Reply:
changed